### ISSUE TITLE
expo-router and metro bundler

### ISSUE THREAD

**Issue by odhekar** [gjtiquia/react-native-web-skia#2]
This is great. I was wondering if you have managed to get this working with expo-router?

**Comment by gjtiquia** [comment]
Glad you found it useful!

I haven't used `expo-router` before, I think it's still rather new. According to the [docs](https://docs.expo.dev/guides/routing-and-navigation/) it's built on top of React Navigation. I'm not sure what that means exactly, but the examples shown in the React Native Skia repo use React Navigation, so I think it should be theoretically supported. Keep in mind, it seems to me that the examples in the React Native Skia repo use the React Native CLI instead of Expo, further testing is needed.

**Comment by odhekar** [comment]
so got everything wired with expo-router as well. It's here for reference,
https://github.com/odhekar/expo-router-react-native-skia

**Comment by gjtiquia** [comment]
This is great! Thanks a lot!

**Comment by odhekar** [comment]
don't want to nag you but were you able to load skia on both web and mobile (Android using expo go)? It loads just blank on expo go but works alright on web. I am using Platform.OS to selectively load skia using WithSkiaWeb (web) or just use the HelloWorld/Breath components as is (Android).

**Comment by gjtiquia** [comment]
Yup I got it to work! Tried so many things, including `Platform.OS`, and I found that the only way to get it to work was with [Platform-specific extensions](https://reactnative.dev/docs/platform-specific-code#platform-specific-extensions). 

If you look into my source code in the /src directory you can see that I used `Main.tsx` for web and `Main.native.tsx` for iOS and Android. 

If you dig deep into the React Native Skia repo under the /example directory, you can see that they used platform-specific extensions too, `index.js` and `index.web.js`. Worth noting that they are using [defered component registration](https://shopify.github.io/react-native-skia/docs/getting-started/web#using-defered-component-registration) instead of code-splitting using the `<WithSkiaWeb>` component.

My guess is that on mobile there would be a compile error when trying to import the `WithSkiaWeb`. Even if you used `Platform.OS`, since it is in the import statement it will still get compiled. The only workaround I know to make sure it doesn't even get compiled is by using Platform-specific extensions.

Check out my answer on [stack overflow](https://stackoverflow.com/questions/76441855/using-expo-with-react-native-skia) where I answered something similar to this.

Wish you luck!
